Analysis/Tasks/PWGHF/taskDDbarCorrelation.cxx

```diff
@@ -0,0 +1,395 @@
+     {"hDeltaPtDDbarSidebands", "D,Dbar candidates sidebands;#it{p}_{T}^{Dbar}-#it{p}_{T}^{D};entries", {HistType::kTH1F, {{144, -36., 36.}}}},
+     {"hDeltaPtMaxMinSidebands", "D,Dbar candidates sidebands;#it{p}_{T}^{max}-#it{p}_{T}^{min};entries", {HistType::kTH1F, {{72, 0., 36.}}}}}};
+
+  //pT ranges for correlation plots: the default values are those embedded in hf_cuts_d0_topik (i.e. the mass pT bins), but can be redefined via json files
+  Configurable<std::vector<double>> binsCorrelations{"ptBinsForCorrelations", std::vector<double>{o2::analysis::hf_cuts_d0_topik::pTBins_v}, "pT bin limits for correlation plots"};
+  //signal and sideband region edges, to be defined via json file (initialised to empty)
+  Configurable<std::vector<double>> signalRegionInner{"signalRegionInner", std::vector<double>(), "Inner values of signal region vs pT"};
+  Configurable<std::vector<double>> signalRegionOuter{"signalRegionOuter", std::vector<double>(), "Outer values of signal region vs pT"};
+  Configurable<std::vector<double>> sidebandLeftInner{"sidebandLeftInner", std::vector<double>(), "Inner values of left sideband vs pT"};
+  Configurable<std::vector<double>> sidebandLeftOuter{"sidebandLeftOuter", std::vector<double>(), "Outer values of left sideband vs pT"};
+  Configurable<std::vector<double>> sidebandRightInner{"sidebandRightInner", std::vector<double>(), "Inner values of right sideband vs pT"};
+  Configurable<std::vector<double>> sidebandRightOuter{"sidebandRightOuter", std::vector<double>(), "Outer values of right sideband vs pT"};
+
+  void init(o2::framework::InitContext&)
+  {
+    // redefinition of pT axes for THnSparse holding correlation entries
+    int nBinspTaxis = binsCorrelations->size() - 1;
+    const double* valuespTaxis = binsCorrelations->data();
+
+    for (int i = 2; i <= 3; i++) {
+      registry.get<THnSparse>(HIST("hMass2DCorrelationPairs"))->GetAxis(i)->Set(nBinspTaxis, valuespTaxis);
+      registry.get<THnSparse>(HIST("hCorrel2DVsPtSignalRegion"))->GetAxis(i)->Set(nBinspTaxis, valuespTaxis);
+      registry.get<THnSparse>(HIST("hCorrel2DVsPtSidebands"))->GetAxis(i)->Set(nBinspTaxis, valuespTaxis);
+    }
+  }
+
+  void process(aod::DDbarPairFull const& pairEntries)
+  {
+    for (auto& pairEntry : pairEntries) {
+      //define variables for widely used quantities
+      double deltaPhi = pairEntry.deltaPhi();
+      double deltaEta = pairEntry.deltaEta();
+      double ptD = pairEntry.ptD();
+      double ptDbar = pairEntry.ptDbar();
+      double massD = pairEntry.mD();
+      double massDbar = pairEntry.mDbar();
+
+      //reject entries outside pT ranges of interest
+      double minPtAllowed = binsCorrelations->at(0);
+      double maxPtAllowed = binsCorrelations->at(binsCorrelations->size() - 1);
+      if (ptD < minPtAllowed || ptDbar < minPtAllowed || ptD > maxPtAllowed || ptDbar > maxPtAllowed) {
+        continue;
+      }
```

Comment:
Isn't this what the `findBin` function does?

Reply:
Modified exploiting the findbin return value